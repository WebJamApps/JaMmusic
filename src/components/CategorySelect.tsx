import {
  FormControl, InputLabel, MenuItem, Select,
} from '@mui/material';
import { Isong } from 'src/providers/Data.provider';

export function CategorySelect({ songJson, setFunc }:
{ songJson: Isong, setFunc: (arg0: Isong) => void }) {
  return (
    <FormControl fullWidth sx={{ marginBottom: '12px' }}>
      <InputLabel id="select-category-label">Category</InputLabel>
      <Select
        style={{ marginBottom: '12px' }}
        labelId="select-category-label"
        id="select-category"
        value={songJson.category}
        label="Category"
        onChange={(evt) => {
          const { target: { value } } = evt;
          setFunc({ ...songJson, category: value });
          return value;
        }}
      >
        <MenuItem value="original">original</MenuItem>
        <MenuItem value="mission">mission</MenuItem>
        <MenuItem value="pub">pub</MenuItem>
      </Select>
    </FormControl>
  );
}
