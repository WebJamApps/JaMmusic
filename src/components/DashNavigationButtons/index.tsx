import React from 'react';
import type { MusicDashboard } from '../../containers/MusicDashboard';

type PageProps = { comp:MusicDashboard };

const DashButtons = ({ comp }:PageProps): JSX.Element => (
  <div
    className="Nav-Buttons"
    id="Nav-Buttons"
    style={{
      display: 'inline',
    }}
  >
    <button
      className="floatRight1"
      type="button"
      id="Photos-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Photos
    </button>
    <button
      className="floatRight2"
      type="button"
      id="Tours-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Tours
    </button>
    <button
      className="floatRight3"
      type="button"
      id="Songs-Button"
      style={{ margin: '5px', fontSize: '55%' }}
      onClick={comp.handleNavClick}
    >
      Songs
    </button>
  </div>
);

export const DashNavigationButtons = ({ comp }:PageProps): JSX.Element => (
  <nav className="white">
    <div
      className="center"
      style={{
        textAlign: 'center',
      }}
    >
      <h4
        className="Musicdash-Title"
        style={{
          margin: '14px', fontWeight: 'bold', color: 'black',
        }}
      >
        Music Dashboard
        {DashButtons({ comp })}
      </h4>
    </div>
  </nav>
);
export default DashNavigationButtons;
