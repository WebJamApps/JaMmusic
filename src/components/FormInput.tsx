import { TextField } from '@mui/material';

interface IformInputProps {
  type: string | undefined,
  label: string, isRequired: boolean | undefined,
  onChange: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined,
  value: string, style?: any
}
export const FormInput = (props:IformInputProps): JSX.Element => {
  const { label, style, isRequired, type, onChange, value } = props;
  let fId = label.toLowerCase();
  fId = fId.replace(/\s/g, '');
  const fIdArr = fId.split('(');
  // eslint-disable-next-line prefer-destructuring
  fId = fIdArr[0];
  return (
        <TextField
            style={style}
            id={fId}
            placeholder={isRequired ? '* '.concat(label) : label}
            type={type}
            name={fId}
            onChange={onChange}
            value={value || ''}
        />
  );
};
