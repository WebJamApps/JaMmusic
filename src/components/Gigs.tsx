import React, { useContext } from 'react';
import { DataGrid, GridColumns, GridEnrichedColDef, GridRenderCellParams } from '@mui/x-data-grid';
import { GigsContext } from '../providers/Gigs.provider';
import HtmlReactParser from 'html-react-parser';
import './Gigs.scss';

export const makeVenue = ():GridEnrichedColDef => {
  return (
    {
      field: 'venue',
      headerName: 'Venue',
      width: 600,
      editable: false,
      renderCell:(params:GridRenderCellParams) => <span>{HtmlReactParser(params.value)}</span>,
    }
  );
};

const columns:GridColumns = [
  {
    field: 'date',
    headerName: 'Date',
    width: 150,
    editable: false,
  },
  {
    field: 'time',
    headerName: 'Time',
    width: 150,
    editable: false,
  },
  {
    field: 'location',
    headerName: 'Location',
    width: 150,
    editable: false,
  },
  makeVenue(),
  {
    field: 'tickets',
    headerName: 'Tickets',
    width: 140,
    editable: false,
  },
];

export const Gigs = (): JSX.Element => {
  const { gigs } = useContext(GigsContext);
  return (
    <div style={{ margin: 'auto', padding:'10px', width:'100%' }}>
      <h4 style={{ textAlign:'center' }}>Gigs</h4>
      <div style={{ height: 400, width: '100%' }}>
      <DataGrid
        rows={gigs}
        columns={columns}
        pageSize={5}
        rowsPerPageOptions={[5]}
        disableSelectionOnClick
      />
      </div>
    </div>
  );
};
