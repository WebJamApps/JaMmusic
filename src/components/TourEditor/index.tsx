import React from 'react';
import AddTime from '../../containers/MusicDashboard/AddTime';
import type { MusicDashboard } from '../../containers/MusicDashboard';
import Ttable from '../TourTable/TourTable_old';
import Utils from './tourEditorUtils';

export interface IeditTour {
  date?:string; time?:string;
  tickets?:string; more?:string; venue?:string; location?:string; _id?:string; datetime?:string
}
type PageProps = { editTour:IeditTour, comp:MusicDashboard };

export const newTourForm = (comp:MusicDashboard, editTour:IeditTour): JSX.Element => {
  let {
    location, tickets, more, date, time, venue,
  } = comp.state;

  date = comp.fixDate(date, editTour);
  if (time === '' && editTour.time !== undefined) { time = editTour.time; }
  if (tickets === '' && editTour.tickets !== undefined) { tickets = editTour.tickets; }
  if (more === '' && editTour.more !== undefined) { more = editTour.more; }
  if (venue === '' && editTour.venue !== undefined) { venue = editTour.venue; }
  if (location === '' && editTour.location !== undefined) { location = editTour.location; }
  return (
    <div className="material-content elevation3" style={{ maxWidth: '9.1in', margin: 'auto' }}>
      <h5 style={{ textAlign: 'center', marginBottom: '30px' }}>
        {editTour._id ? 'Edit ' : 'Create a New '}
        Tour Event
      </h5>
      <p>{' '}</p>
      <form id="new-tour" style={{ marginLeft: '4px', marginTop: '12px' }}>
        <p>* Date</p>
        {comp.forms.makeInput('date', 'Date', true, comp.onChange, date)}
        <AddTime setFormTime={comp.setFormTime} initTime={time} show={true}/>
        {Utils.editor(venue, comp)}
        <p>{' '}</p>
        {comp.forms.makeInput('text', 'Location', true, comp.onChange, location)}
        {comp.forms.makeInput('text', 'Tickets', false, comp.onChange, tickets)}
        {comp.forms.makeInput('text', 'More', false, comp.onChange, more)}
        {Utils.tourButtons(comp)}
      </form>
    </div>
  );
};

export const TourEditor = ({ editTour, comp }:PageProps): JSX.Element => (
  <div className="Tour-Block">
    <p>&nbsp;</p>
    {newTourForm(comp, editTour)}
    <p>&nbsp;</p>
    {!editTour._id ? (
      <div className="search-table-outer" style={{ maxWidth: '96%', margin: 'auto', zIndex: 0 }}>
        <h5 style={{ textAlign: 'center', marginBottom: '3px' }}>Modify</h5>
        <Ttable deleteButton />
      </div>
    ) : null}
  </div>
);
