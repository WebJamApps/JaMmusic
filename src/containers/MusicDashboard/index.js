import React, { Component, useState } from 'react';
import moment from 'moment';
import { withRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DateFnsUtils from "@date-io/date-fns";
import {
  DatePicker,
  MuiPickersUtilsProvider,
  KeyboardTimePicker,
} from '@material-ui/pickers';
import mapStoreToProps from '../../redux/mapStoreToProps';

export class MusicDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect: false,
      date: '',
      time: '',
      tickets: '',
      more: '',
      selectedDate: new Date()
    };
    this.onChange = this.onChange.bind(this);
    this.createTour = this.createTour.bind(this);
    this.validateForm = this.validateForm.bind(this);
    this.createTourApi = this.createTourApi.bind(this);
    this.handleDateChange = this.handleDateChange.bind(this)
  }

  componentDidMount() { document.title = 'Music Dashboard | Web Jam LLC'; }

  handleDateChange(e) {
    this.setState({ selectedDate: e })
    console.log(e)
  }
  onChange(evt) {
    evt.preventDefault();
    this.setState({ [evt.target.id]: evt.target.value });
  }

  validateForm() {
    const {
      date, time, location, venue,
    } = this.state;
    if (date && time && location && venue && date !== '') return false;
    return true;
  }

  createTourApi(tour1) { // eslint-disable-line class-methods-use-this
    const { scc, auth } = this.props;
    const tour = tour1;
    tour.datetime = new Date(tour.date);
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.emit('newTour', { tour, token: auth.token });
    this.setState({ redirect: true });
  }

  createTour() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const tour = {
      date, time, location, venue, tickets, more,
    };
    return this.createTourApi(tour);
  }

  makeInput(type, name, isRequired, onChange, value) { // eslint-disable-line class-methods-use-this
    return (
      <label htmlFor={name} style={{ marginTop: '14px', paddingTop: 0 }}>
        {isRequired ? '* ' : ''}
        {name[0].toUpperCase() + name.slice(1)}
        <br />
        <input id={name} type={type} name={name} onChange={onChange} required={isRequired} value={value} />
      </label>
    );
  }

  newTourForm(date, time, buttonStyle) {
    const {
      venue, location, tickets, more,
    } = this.state;
    return (
      <form id="new-tour" style={{ marginTop: '4px', paddingLeft: '10px' }}>
        {this.makeInput('date', 'date', true, this.onChange, date)}
        {this.makeInput('text', 'time', true, this.onChange, time)}
        {this.makeInput('text', 'venue', true, this.onChange, venue)}
        {this.makeInput('text', 'location', true, this.onChange, location)}
        {this.makeInput('text', 'tickets', false, this.onChange, tickets)}
        {this.makeInput('text', 'more', false, this.onChange, more)}
        <div  className = "divStyle">
          <MuiPickersUtilsProvider utils={DateFnsUtils} >
            <KeyboardTimePicker
              id="time-picker"
              label="Time picker"
              value={this.selectedDate}
              onChange={this.handleDateChange}
              KeyboardButtonProps={{ 'aria-label': 'change time' }}
            />
          </MuiPickersUtilsProvider>
        </div>
        <div style={{ textAlign: 'right', marginTop: '10px', maxWidth: '85%' }}>
          <span style={{ fontSize: '16px', marginRight: '38%', fontFamily: 'Habibi' }}><i>* Required</i></span>
          <button style={buttonStyle} disabled={this.validateForm()} type="button" onClick={this.createTour}>Create Tour</button>
        </div>
      </form>
    );
  }


  render() {
    const {
      date, time, buttonStyle, redirect,
    } = this.state;
    return (
      <div className="page-content" >
        {redirect ? <Redirect to="/music" /> : null}
        <h3 style={{ textAlign: 'center', margin: '14px', fontWeight: 'bold' }}>Music Dashboard</h3>
        <h5 style={{ textAlign: 'center', marginBottom: 0 }}>Create a New Tour</h5>
        {this.newTourForm(date, time, buttonStyle)}
      </div>
    );
  }
}
MusicDashboard.propTypes = { scc: PropTypes.shape({ emit: PropTypes.func }), auth: PropTypes.shape({ token: PropTypes.string }).isRequired };
MusicDashboard.defaultProps = { scc: { emit: () => { } } };
export default withRouter(connect(mapStoreToProps)(MusicDashboard));
