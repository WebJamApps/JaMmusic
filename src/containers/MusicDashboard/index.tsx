import React, { Component } from 'react';
import { Editor } from '@tinymce/tinymce-react';
import moment from 'moment';
import { withRouter, Redirect, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import mapStoreToProps from '../../redux/mapStoreToProps';
import forms from '../../lib/forms';
import commonUtils from '../../lib/commonUtils';
import AddTime from '../../lib/timeKeeper';
import Ttable from '../../components/tour-table';

interface MusicDashboardProps extends RouteComponentProps<any> {
  dispatch: (...args: any) => any;
  scc: { transmit: (...args: any[]) => any };
  auth: { token: string };
  editTour: { date?: string; time?: string; tickets?: string; more?: string; venue?: string; location?: string; _id?: string; datetime?: string };
}
type MusicDashboardState = {
  location: string;
  venue: any | string;
  redirect: boolean;
  date: string;
  time: string;
  tickets: string;
  more: string;
  [x: number]: any;
};
export class MusicDashboard extends Component<MusicDashboardProps, MusicDashboardState> {
  forms: any;

  commonUtils: any;

  constructor(props: any) {
    super(props);
    this.state = {
      redirect: false, date: '', time: '', tickets: '', more: '', venue: '', location: '',
    };
    this.forms = forms;
    this.onChange = this.onChange.bind(this);
    this.createTour = this.createTour.bind(this);
    this.validateForm = this.validateForm.bind(this);
    this.createTourApi = this.createTourApi.bind(this);
    this.handleEditorChange = this.handleEditorChange.bind(this);
    this.setFormTime = this.setFormTime.bind(this);
    this.commonUtils = commonUtils;
    this.checkEdit = this.checkEdit.bind(this);
    this.editTourAPI = this.editTourAPI.bind(this);
    this.resetEditForm = this.resetEditForm.bind(this);
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Music Dashboard'); }

  onChange(evt: any) {
    const { editTour } = this.props;
    if (editTour.venue !== undefined) this.checkEdit();
    this.setState({ [evt.target.id]: evt.target.value });
  }

  setFormTime(data: any) {
    this.setState({ time: data });
  }

  checkEdit() {
    let {
      date, time, tickets, more, venue, location,
    } = this.state;
    const { editTour, dispatch } = this.props;
    if (date === '' && editTour.date !== undefined) { date = editTour.datetime.split('T')[0]; }//eslint-disable-line
    if (time === '' && editTour.time !== undefined) { time = editTour.time; }
    if (tickets === '' && editTour.tickets !== undefined) { tickets = editTour.tickets; }
    if (more === '' && editTour.more !== undefined) { more = editTour.more; }
    if (venue === '' && editTour.venue !== undefined) { venue = editTour.venue; }
    if (location === '' && editTour.location !== undefined) { location = editTour.location; }
    this.setState({
      date, time, tickets, more, venue, location,
    });

    if (editTour.venue !== undefined) {
      dispatch({ type: 'EDIT_TOUR', data: { _id: editTour._id } });
    }
  }

  resetEditForm(evt: { preventDefault: (...args: any) => any }) {
    evt.preventDefault();
    const { dispatch } = this.props;
    dispatch({ type: 'EDIT_TOUR', data: {} });
    this.setState({
      date: '', time: '', tickets: '', more: '', venue: '', location: '',
    });
  }

  handleEditorChange(venue: any) { this.setState({ venue }); }

  validateForm() {
    const {
      date, time, location, venue,
    } = this.state;
    if (date && time && location && venue && date !== '') return false;
    return true;
  }

  createTourApi(tour1: any) {
    const { scc, auth } = this.props;
    const tour = tour1;
    tour.datetime = tour.date;
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.transmit('newTour', { tour, token: auth.token });
    this.setState({ redirect: true });
    return true;
  }

  createTour() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const tour = {
      date, time, location, venue, tickets, more,
    };
    return this.createTourApi(tour);
  }

  editor(venue: string) {
    return (
      <Editor
        value={venue}
        apiKey={process.env.TINY_KEY}
        init={{
          height: 400,
          menubar: 'insert',
          selector: 'textarea',
          menu: { format: { title: 'Format', items: 'forecolor backcolor' } },
          plugins: [
            'advlist autolink lists link image charmap print preview anchor',
            'searchreplace visualblocks code fullscreen',
            'insertdatetime media table paste code help wordcount',
          ],
          toolbar:
            'undo redo | formatselect | bold italic backcolor forecolor |'
            + 'alignleft aligncenter alignright alignjustify |'
            + 'bullist numlist outdent indent | removeformat | help',
        }}
        onEditorChange={this.handleEditorChange}
      />
    );
  }

  editTourAPI() {
    const {
      date, time, location, venue, tickets, more,
    } = this.state;
    const { editTour, scc, auth } = this.props;
    const tour = {
      date, time, location, venue, tickets, more, datetime: date,
    };
    const m = moment(tour.date, 'YYYY-MM-DD');
    tour.date = m.format('ll');
    scc.transmit('editTour', { tour, token: auth.token, tourId: editTour._id });
    this.resetEditForm({ preventDefault: () => { } });
    this.setState({ redirect: true });
    return true;
  }

  tourButtons() {
    const { editTour } = this.props;
    return (
      <div style={{ textAlign: 'left', marginTop: '10px', maxWidth: '85%' }}>
        <span style={{
          fontSize: '16px', marginRight: '20px', fontFamily: 'Habibi', position: 'relative', display: 'inline-block',
        }}
        >
          <i>* Required</i>
        </span>
        {editTour._id ? (
          <button className="floatRight" type="button" id="cancel-edit-pic" onClick={this.resetEditForm}>
            Cancel
          </button>
        ) : null}
        <button
          className="floatRight"
          disabled={this.validateForm()}
          type="button"
          onClick={editTour._id ? this.editTourAPI : this.createTour}
        >
          {editTour._id ? 'Edit' : 'Create'}
          {' '}
          Tour
        </button>
      </div>
    );
  }

  newTourForm() {
    let {
      location, tickets, more, date, time, venue,
    } = this.state;
    const { editTour } = this.props;
    if (editTour.date !== undefined) { date = editTour.datetime.split('T')[0]; }//eslint-disable-line
    if (editTour.time !== undefined) { time = editTour.time; }
    if (editTour.tickets !== undefined) { tickets = editTour.tickets; }
    if (editTour.more !== undefined) { more = editTour.more; }
    if (editTour.venue !== undefined) { venue = editTour.venue; }
    if (editTour.location !== undefined) { location = editTour.location; }
    return (
      <form id="new-tour" style={{ marginLeft: '4px', marginTop: '4px' }}>
        {this.forms.makeInput('date', 'Date', true, this.onChange, date)}
        <AddTime setFormTime={this.setFormTime} initTime={time} />
        <div className="horiz-scroll">
          <div style={{ width: '850px', margin: 'auto' }}>
            <p style={{ marginBottom: 0 }}>* Venue</p>
            {this.editor(venue)}
          </div>
        </div>

        {this.forms.makeInput('text', 'Location', true, this.onChange, location)}
        {this.forms.makeInput('text', 'Tickets', false, this.onChange, tickets)}
        {this.forms.makeInput('text', 'More', false, this.onChange, more)}
        {this.tourButtons()}
      </form>
    );
  }

  render() {
    const { redirect } = this.state;
    const { editTour } = this.props;
    return (
      <div className="page-content">
        {redirect ? <Redirect to="/music" /> : null}
        <h3 style={{ textAlign: 'center', margin: '14px', fontWeight: 'bold' }}>Music Dashboard</h3>
        <div className="material-content elevation3" style={{ maxWidth: '9.1in', margin: 'auto' }}>
          <h5 style={{ textAlign: 'center', marginBottom: 0 }}>
            {editTour._id ? 'Edit' : 'Create a New'}
            {' '}
            Tour Event
          </h5>
          {this.newTourForm()}
        </div>
        <p>&nbsp;</p>
        {!editTour._id ? (
          <div className="material-content elevation3" style={{ maxWidth: '10in', margin: 'auto' }}>
            <h5 style={{ textAlign: 'center', marginBottom: 0 }}>Modify Event</h5>
            <Ttable deleteButton />
          </div>
        ) : null}
        <p>&nbsp;</p>
        <p>&nbsp;</p>
      </div>
    );
  }
}

export default withRouter(connect(mapStoreToProps, null)(MusicDashboard));

