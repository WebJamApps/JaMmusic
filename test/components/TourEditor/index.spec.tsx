/* eslint-disable @typescript-eslint/no-explicit-any */

import { shallow } from 'enzyme';
import type { Auth } from 'src/redux/mapStoreToProps';
import { TourEditor, newTourForm } from '../../../src/components/TourEditor';
import { MusicDashboard } from '../../../src/containers/MusicDashboard';

describe('TourEditor', () => {
  it('is defined', () => {
    expect(TourEditor).toBeDefined();
  });
  let wrapper: any;
  const auth = { token: '' } as Auth;
  const anyProp: any = {};
  const scc:any = { transmit: () => { } };
  beforeEach(() => {
    wrapper = shallow<MusicDashboard>(<MusicDashboard
      auth={auth as Auth}
      scc={scc}
      dispatch={jest.fn()}
      editTour={anyProp}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editSong={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
  });
  it('calls on change', () => {
    wrapper.instance().checkEdit = jest.fn();
    wrapper.instance().setState = jest.fn();
    const evt:any = { preventDefault: () => { }, persist: jest.fn(), target: { id: 'hi', value: 11 } };
    wrapper.instance().onChange(evt);
    expect(wrapper.instance().setState).toHaveBeenCalled();
  });
  it('handles onChange with editTour', () => {
    const wrapper2 = shallow<MusicDashboard>(
      <MusicDashboard
        auth={auth as Auth}
        scc={scc}
        dispatch={jest.fn()}
        editTour={{ venue: 'wjllc' }}
        history={anyProp}
        location={anyProp}
        match={anyProp}
        editSong={anyProp}
        editPic={anyProp}
        images={anyProp}
        showTable={anyProp}
      />,
    );
    wrapper2.instance().checkEdit = jest.fn();
    const evt:any = { preventDefault: () => { }, persist: jest.fn(), target: { id: 'hi', value: '11' } };
    wrapper2.instance().onChange(evt);
    expect(wrapper2.instance().checkEdit).toHaveBeenCalled();
  });
  it('calls handleEditorChange', () => {
    wrapper.instance().checkEdit = jest.fn();
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.venue).toBe('<p>Cool Venue</p>'); });
    wrapper.instance().handleEditorChange('<p>Cool Venue</p>');
  });
  it('calls the tour API', () => {
    wrapper.instance().setState = jest.fn((obJ) => { expect(obJ.redirect).toBe(true); });
    wrapper.instance().createTourApi({ date: '2019-10-10' });
  });
  it('returns the validation', () => {
    wrapper.setState({
      date: '10-24-2019',
      time: '10:00',
      location: 'A place',
      venue: 'Venue',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(false);
  });
  it('creates the tour', () => {
    wrapper.setState({
      date: '10-24-2019',
      time: '10:00',
      location: 'A place',
      venue: 'Venue',
      tickets: '1',
      more: 'yes',
    });
    const result = wrapper.instance().createTour();
    expect(result).toBe(true);
  });
  it('resets the edit form', () => {
    wrapper.instance().setState = jest.fn((obj) => expect(obj.date).toBe(''));
    wrapper.update();
    wrapper.instance().resetEditForm({ preventDefault: () => { } });
  });
  it('calls checkEdit', () => {
    wrapper.instance().checkEdit('<p>Cool Venue</p>');
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.venue).toBe('<p>Cool Venue</p>'); });
  });
  it('calls setFormTime', () => {
    wrapper.instance().setFormTime('12:00 pm');
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.time).toBe('12:00 pm'); });
  });
  it('setSongState when editSong.composer', () => {
    const song:any = { composer: 'Josh' };
    wrapper.instance().setSongState(song);
    expect(wrapper.instance().state.songState.composer).toBe('Josh');
  });
  it('checks edit when editTour', () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editTour={{
        datetime: '2020-10-10T000',
        _id: '123',
        date: '2020-10-10',
        time: '5 pm',
        tickets: 'free',
        more: '',
        venue: 'beer garden',
        location: 'salem, va',
      }}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editSong={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    wrapper2.instance().setState = jest.fn();
    const sO = {
      date: '2020-10-10', time: '5 pm', tickets: 'free', more: '', venue: 'beer garden', location: 'salem, va',
    };
    wrapper2.instance().checkEdit();
    expect(wrapper2.instance().setState).toHaveBeenCalledWith(sO);
  });
  it('checks edit when not editTour', () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editTour={{}}
      editSong={anyProp}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    wrapper2.instance().setState = jest.fn();
    const sO = {
      date: '', time: '', tickets: '', more: '', venue: '', location: '',
    };
    wrapper2.instance().checkEdit();
    expect(wrapper2.instance().setState).toHaveBeenCalledWith(sO);
  });
  it('resets edit form when editTour', () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editSong={anyProp}
      editTour={{
        datetime: '2020-10-10T000',
        _id: '123',
        date: '2020-10-10',
        time: '5 pm',
        tickets: 'free',
        more: '',
        venue: 'beer garden',
        location: 'salem, va',
      }}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    wrapper2.setState({ venue: 'beer garden' });
    wrapper2.instance().setState = jest.fn();
    const evt:any = { preventDefault: () => { } };
    wrapper2.instance().resetEditForm(evt);
    const sO = {
      date: '', time: '', tickets: '', more: '', venue: '', location: '',
    };
    expect(wrapper2.instance().setState).toHaveBeenCalledWith(sO);
  });
  it('calls edit tour api', async () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editSong={anyProp}
      editTour={{
        datetime: '2020-10-10T000',
        _id: '123',
        date: '2020-10-10',
        time: '5 pm',
        tickets: 'free',
        more: '',
        venue: 'beer garden',
        location: 'salem, va',
      }}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    wrapper2.setState({
      date: '2020-10-10',
      time: '5 pm',
      tickets: 'free',
      more: '',
      venue: 'beer garden',
      location: 'salem, va',
    });
    const r = wrapper2.instance().editTourAPI();
    expect(r).toBe(true);
  });
  it('Tour Block is defined', () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editSong={anyProp}
      editTour={{
        datetime: '2020-10-10T000',
        _id: '123',
        date: '2020-10-10',
        time: '5 pm',
        tickets: 'free',
        more: '',
        venue: 'beer garden',
        location: 'salem, va',
      }}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    const instance = wrapper2.instance();
    instance.setState({
      date: '2020-10-10',
      time: '5 pm',
      tickets: 'free',
      more: '',
      venue: 'beer garden',
      location: 'salem, va',
    });
    const wrapper3 = shallow(<TourEditor comp={instance} editTour={instance.props.editTour} />);
    expect(wrapper3.find('.search-outer-table')).toBeDefined();
  });
  it('reaches Tour table delete button', () => {
    const wrapper2 = shallow<MusicDashboard>(<MusicDashboard
      auth={auth}
      scc={scc}
      dispatch={jest.fn()}
      editSong={anyProp}
      editTour={{
        datetime: '',
        _id: '',
        date: '',
        time: '',
        tickets: '',
        more: '',
        venue: '',
        location: '',
      }}
      history={anyProp}
      location={anyProp}
      match={anyProp}
      editPic={anyProp}
      images={anyProp}
      showTable={anyProp}
    />);
    const instance = wrapper2.instance();
    instance.setState({
      date: '',
      time: '',
      tickets: '',
      more: '',
      venue: '',
      location: '',
    });
    const wrapper3 = shallow(<TourEditor comp={instance} editTour={instance.props.editTour} />);
    expect(wrapper3.find('.search-outer-table')).toBeDefined();
  });
  it('newTourForm renders when editTour is missing some attributes', () => {
    const comp:any = {
      state: { location: '' },
      validateForm: jest.fn(),
      fixDate: jest.fn(),
      forms: { makeInput: jest.fn() },
      props: {
        auth,
        scc,
        dispatch: jest.fn(),
        editSong: anyProp,
        editTour: {
          datetime: '',
          _id: '',
          date: '',
          time: '',
          tickets: '',
          more: '',
          venue: '',
          location: '',
        },
        history: anyProp,
        location: anyProp,
        match: anyProp,
      },
    };
    const form = shallow(newTourForm(comp, { _id: '123' }));
    expect(form.find('div.elevation3').exists()).toBe(true);
  });
});
