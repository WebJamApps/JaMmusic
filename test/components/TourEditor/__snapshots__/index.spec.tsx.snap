// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourEditor renders correctly 1`] = `ShallowWrapper {}`;
